### PR TITLE
Add Sysprop to Adjust IO Buffer Size

### LINKED ISSUE #46803: NativeRealmIntegTests tests failed with OutOfMemoryError Direct buffer memory
All the tests in `NativeRealmIntegTests` failed today on `7.x` branch:
https://gradle-enterprise.elastic.co/s/q3qkmolurbooa

Something went wrong when allocating direct buffer memory in the Transport layer:
```java
WARNING: Uncaught exception in thread: Thread[Thread-51,5,TGRP-NativeRealmIntegTests]
java.lang.OutOfMemoryError: Direct buffer memory
	at __randomizedtesting.SeedInfo.seed([7168F60F3A169301]:0)
	at java.nio.Bits.reserveMemory(Bits.java:694)
	at java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:123)
	at java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:311)
	at org.elasticsearch.transport.CopyBytesSocketChannel.lambda$static$0(CopyBytesSocketChannel.java:65)
	at java.lang.ThreadLocal$SuppliedThreadLocal.initialValue(ThreadLocal.java:284)
	at java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:180)
	at java.lang.ThreadLocal.get(ThreadLocal.java:170)
	at org.elasticsearch.transport.CopyBytesSocketChannel.getIoBuffer(CopyBytesSocketChannel.java:141)
	at org.elasticsearch.transport.CopyBytesSocketChannel.doReadBytes(CopyBytesSocketChannel.java:121)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:597)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:551)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:748)
```

It does not reproduce locally:

```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.authc.esnative.NativeRealmIntegTests.testUpdatingUserAndAuthentication" -Dtests.seed=7168F60F3A169301 -Dtests.security.manager=true -Dtests.locale=mk-MK -Dtests.timezone=Asia/Damascus -Dcompiler.java=12 -Druntime.java=8
```

### PR DESCRIPTION
**Pull request by original-brownbear**
The 1MB IO-buffer size per transport thread is causing trouble in
some tests, albeit at a low rate. Reducing the number of transport
threads was not enough to fully fix this situation.
Allowing to configure the size of the buffer (via an undocumented system property)
and reducing it by more than an order of magnitude should fix these tests for good.

Closes #46803

### CODE REVIEW

**Comment by tbrooks8** [comment]
Is the `MaxDirectMemorySize` system property set when these tests run?

**Comment by original-brownbear** [comment]
@tbrooks8 nah it's not afaik. So we should be getting the same 512M we get for `-Xmx` as native memory. Could up that, but I'm not sure if we want to push things too much on that end with all the massive test parallelization we have now? (we may have to push it to like 1G but that might make all test runs take more memory since more is available) => this seemed like the safer bet?

File: x-pack/plugin/security/build.gradle [hunk 1]
```diff
@@ -306,6 +306,12 @@ test {
      * other if we allow them to set the number of available processors as it's set-once in Netty.
      */
     systemProperty 'es.set.netty.runtime.available.processors', 'false'
+
+    /*
+     * Some tests in this module set up a lot of transport threads so we reduce the buffer size per transport thread from the 1M default
+     * to keep direct memory usage under control.
+     */
+    systemProperty 'es.transport.buffer.size', '64k'
```
Comment (tbrooks8) [inline x-pack/plugin/security/build.gradle]: Can we start at 256KB and see if it is still happening? That is what nio transport has and seems to never trigger this.

Comment (original-brownbear) [reply x-pack/plugin/security/build.gradle]: Sure done :)

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Tim!

**Pull request merged**